File: Java/O.java
// Компилятор языка "О"
public class O {

static void Init() {
   Text.Reset();
   if( !Text.Ok )
      Error.Message(Text.Message);
   Scan.Init();
   Gen.Init();
}

static void Done() {
   Text.Close();
}

public static void main(String[] args) {
   System.out.println("\nКомпилятор языка О");
   if( args.length == 0 )
      Location.Path = null;
   else
      Location.Path = args[0];
   Init();         // Инициализация
   Pars.Compile(); // Компиляция   
   OVM.Run();      // Выполнение   
   Done();         // Завершение   
}

}

File: CompO_dev/o.c
/* Компилятор языка О (o.c) */

#include <string.h>
#include <stdio.h>

#include "text.h"
#include "scan.h"
#include "ovm.h"
#include "scan.h"
#include "gen.h"
#include "location.h"
#include "pars.h"
#include "error.h"
#include "ovmtobin.h"

void Init(void) {
	ResetText();
	if( ResetError )
		Error(Message);
	InitScan();
	InitGen();
}

void Done(void) {
	CloseText();
}

int main(int argc, char *argv[]) {
	puts((char*)"\nКомпилятор языка О");
	if( argc <= 1 )
		Path = NULL;
	else
		Path = argv[1];
	Init();    /* Инициализация */
	Compile(); /* Компиляция    */
	Test_Out_M();
	Run();     /* Выполнение    */
	Done();    /* Завершение    */
	return 0;
}

